### PR TITLE
Split deployment between tenant and infra cluster

### PR DESCRIPTION
**Pull request by hdurand0710**
Fixes: #4 

This PR is the kubevirt-csi-operator part of the split, updated to only deploy the node (the controller will be deployed in the seed-controller) and needed configuration (RBAC,...).

Signed-off-by: Helene Durand <helene@kubermatic.com>

### CODE REVIEW

**Review by mfranczy — COMMENTED** [review]
Can you confirm if during upgrade the CSI driver controller and secrets will be gone? Since the operator `Owns` the deployment and secrets I expect this will happen.

**Comment by hdurand0710** [comment]
That's a good point.
I confirm that the `Secret`, deployed by the add-on is gone.
But the `Tenant` is migrated to the new version, but still has the same UID, hence the controller is still owned by it.
I've made the choice in to deploy a Tenant `tenant-split-deploy`, and cleanup the `tenant` Tenant in the 0.2.0 version. The Controller is then cascade deleted.
Watch is only (same mechanism) on the `Tenant`, no need to keep watching the controller deployment.
Let me know what you think.

We have a much annoying issue with the move of the CSI driver SA to the dedicated namespace for a migration. I will discuss this with you today.

File: go.mod [hunk 1]
```diff
@@ -3,7 +3,6 @@ module kubevirt-csi-driver-operator
 go 1.19
```
Comment (mfranczy) [inline go.mod:3]: it's my fault, but could you change `module kubevirt-csi-driver-operator` to
`module github.com/kubermatic/kubevirt-csi-driver-operator`?

**Comment by hdurand0710** [comment]
Updated with the latest changes we talked about:
- same Tenant
- clean unused Deployment

**Review by mfranczy — APPROVED** [review]

**Pull request merged**